smarty/sampler_smirky.py

```diff
@@ -683,21 +904,26 @@ def sample_types(self):
 
             # Reject if new type matches nothing
             if proposed_typecounts[new_env.label] == 0:
-                if self.verbose: print("Type '%s' (%s) unused in dataset; rejecting." % (new_env.label, new_env.asSMIRKS()))
+                self.log.write("Type '%s' (%s) unused in dataset; rejecting.\n" % (new_env.label, new_env.asSMIRKS()))
                 self.types_with_no_matches.append(new_env.asSMIRKS())
                 return False
 
             # Reject if parent type is now unused (UNLESS parent is a base type)
             if env.label not in [e.label for e in self.baseTypes]:
                 # parent not in base types
                 if proposed_typecounts[env.label] == 0:
-                    if self.verbose: print("Parent type '%s' (%s) now unused in dataset; rejecting." % (env.label, env.asSMIRKS()))
+                    self.log.write("Parent type '%s' (%s) now unused in dataset; rejecting.\n" % (env.label, env.asSMIRKS()))
                     return False
 
             # updated proposed parent dictionary
             # TODO: update parent dictionary
```

Comment:
You have "TODO" items strewn a few places through this code. Maybe doublecheck occurrences of the word and make sure none should be resolved right now, and then create an issue to resolve the others (tagged as low priority if it is indeed low priority)...